### PR TITLE
feat(validation): probability calibration & ECE with temperature scaling (ML-VAL-002)

### PR DESCRIPTION
**Pull request by Opselon**
## ML-VAL-002 — Probability Calibration & Expected Calibration Error (ECE)

Implements the Guo et al. (2017) calibration toolbox for the 3-class
(NO_TRADE / BUY / SELL) contract and measures the live softmax's calibration gap.

### What it does
- New `src/nexus_scalp/research/calibration.py` — multiclass `compute_ece`,
  `reliability_diagram`, `brier_score`, `log_loss`, `fit_temperature`,
  `TemperatureScaler` (fit / apply_to_logits / apply_to_probs / wrap /
  dict round-trip), `evaluate_temperature_scaling` → `CalibrationReport`.
- Temperature is fitted by **golden-section search on log T** — deterministic,
  no optimizer state, no seed dependence, no GPU. Bounds `T ∈ [0.05, 100]`
  with a fail-loud degenerate-T abort (the task's ABORT_CONDITIONS).
- Binning convention is single-sourced in `_bin_mask()`: half-open `[lo, hi)`
  with a closed top bin, so a perfectly-calibrated flat model lands in a real
  bin and confidence 1.0 always lands top-bin.
- Audit published to `docs/research/CALIBRATION_AUDIT.md`, including an
  inventory of the pre-existing binary Platt subsystem so the multiclass work
  is not mistaken for duplication.

### Headline finding
The live path is **overconfident by construction**:
`InferenceService.infer_probabilities` (`application/live/inference.py:170-185`)
applies raw softmax, and `SignalPolicy` (`signals/policy.py:550-600`)
thresholds it at 0.35 / 0.60 with no calibration correction anywhere. The live
tensor contract is untouched (NON_GOALS honored).

### Verification (run on this branch)
- `pytest tests/unit/test_calibration_research.py` → **63 passed**
- `ruff check` / `ruff format --check` → clean
- `mypy src/nexus_scalp/research/calibration.py --explicit-package-bases` → no issues
- `scripts/ci/verify_critical_suite_manifest.py` → `CRITICAL_SUITE_MANIFEST_OK: 220 paths`

### Benchmark (BENCHMARK_PLAN, 20,000 OOS samples)
```
confidence 0.8972 vs accuracy 0.6934    T = 3.5482
ECE   0.20399 -> 0.00640   (96.9% reduction)
Brier 0.16659 -> 0.13940
NLL   1.23380 -> 0.71445
```
The honest OOS protocol (fit T on train, evaluate on held-out val) is a
separate green test; real NSE folds will show a smaller reduction.

### Test defects found and fixed (not production-code defects)
1. The first fixture derived labels from `argmax(logits)`, making the model
   100% accurate *by construction* — ECE read ~0, T ran to the 0 boundary, and
   every "temperature improves calibration" test was unfalsifiable. Fixtures
   now return the **drawn** labels; a new test asserts the fold is genuinely
   miscalibrated (confidence > 0.80, accuracy < 0.80).
2. A `compute_ece > 0.9` assert was unreachable — the 3-class ceiling is
   `1 - 1/C ≈ 0.667`. Replaced with a ceiling-respecting bound.
3. ECE is not monotone in overconfidence (zero at both extremes); the test now
   asserts the monotone quantity, mean confidence.
4. Brier anchor corrected from `2/3` to `2/9` (`brier_score` averages over all
   `N*C` elements, not per row).

### CODE REVIEW

**Comment by Opselon** [comment]
COORDINATION (release coordinator, 2026-09-21): Your PR #339 was fully GREEN (15 success / 4 tier-skipped, status CLEAN, mergeable) and I was about to squash-merge it, but PR #337 (ML-EXP-003, squash-merged as 83705493) landed first and your PR is now CONFLICTING/DIRTY against the new main. Do not force-push to resolve: merge origin/main (83705493) INTO your branch, resolve append-only (keep all taskboard/ledger rows from both sides — swarm_log.md and agents/taskboard.md are the chronic append-only conflict files; on tests/critical_suite.txt keep BOTH suite blocks), re-run the quick gates (ruff format --check ., ruff check ., and your new test file), push the same branch, and it should go straight back to CLEAN — your checks carry over, only a re-run of the quality job is triggered. Your lane's work looks good; this is purely the multi-agent race at merge time. — ML-PHASE1 release coordinator
